### ISSUE TITLE
Online configuration reload -

### ISSUE THREAD

**Issue by thewacokid** [mar-file-system/marfs#161]
I know we've discussed this in the past, but with new users likely to be added (and their quotas adjusted) on the fly, we can't rely on restarting the FUSE daemon to pick up the new configuration.

Not super urgent *yet*, but there are many situations where we have to interrupt user access to update configs.

**Comment by wfvining** [comment]
Would something like a signal handler for a user-defined signal that re-reads and re-validates the config suffice for this? Admins could update the config and signal FUSE with `kill -s SIGUSR1 <fuse-pid>`. We will need to carefully look at how this might affect the various threads that could be running, it would probably lead to a data race on the global `marfs_config` so we will need to consider whether locking is acceptable around all accesses to the config.

**Comment by jti-lanl** [comment]
Yes, data-race on `marfs_config`.  Fuse has many threads without any state, except what is in file-handles, and marfs_config.  Config changes could potentially change anything -- repo names, mdfs paths, etc.  Even if there isn't a data-dace in some cases, threads may persist in using obsolete info.
